### PR TITLE
Added Klarna support via Checkout API

### PR DESCRIPTION
**Pull request by lancergr**
**Description**
Added Klarna support via Checkout API. RatePay is using /authorise endpoint

**Tested scenarios**
Checkout using: Klarna (SE and DE) and RatePay (DE)

### CODE REVIEW

File: adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java [hunk 1]
```diff
@@ -207,7 +204,7 @@ private void updatePaymentRequest(final String merchantAccount,
             paymentsRequest.setBillingAddress(setAddressData(billingAddress));
             // set PhoneNumber if it is provided
             String phone = billingAddress.getPhone();
-            if (phone != null && !phone.isEmpty()) {
+            if (phone != null && ! phone.isEmpty()) {
```
Comment (cyattilakiss) [inline adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java]: the space can stay I think

File: adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java [hunk 2]
```diff
@@ -227,34 +224,36 @@ private void updatePaymentRequestForCC(PaymentsRequest paymentsRequest, CartData
         String encryptedExpiryMonth = cartData.getAdyenEncryptedExpiryMonth();
         String encryptedExpiryYear = cartData.getAdyenEncryptedExpiryYear();
 
-        if (!StringUtils.isEmpty(encryptedCardNumber) && !StringUtils.isEmpty(encryptedExpiryMonth) && !StringUtils.isEmpty(encryptedExpiryYear)) {
+        if (! StringUtils.isEmpty(encryptedCardNumber) && ! StringUtils.isEmpty(encryptedExpiryMonth) && ! StringUtils.isEmpty(encryptedExpiryYear)) {
```
Comment (cyattilakiss) [inline adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java]: Is it an autoformatter - maybe it's just me but like better without space :D

File: adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java [hunk 3]
```diff
@@ -530,6 +532,96 @@ public void setOpenInvoiceData(PaymentRequest paymentRequest, CartData cartData,
...
+            if (entry.getProduct().getName() != null && ! entry.getProduct().getName().isEmpty()) {
+                description = entry.getProduct().getName();
+            }
+
+            // Tax of total price (included quantity)
+            Double tax = entry.getTaxValues().stream().map(TaxValue::getAppliedValue).reduce(0.0, (x, y) -> x + y);
+
+            // Calculate Tax per quantitiy
+            if (tax > 0) {
+                tax = tax / entry.getQuantity().intValue();
```
Comment (cyattilakiss) [inline adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java]: Can the entry.getQuantity().intValue() be zero in any case?

Comment (lancergr) [reply adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java]: shouldn't but prevented in L560

File: adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java (hunk 2 above)
Comment (lancergr) [reply adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java]: indeed autoformatter

**Pull request merged**